filesystem/fs.go

```diff
@@ -76,11 +76,13 @@ func (cfs CFS) create(parentID uint64, name string, node *Node) error {
 	if err != nil {
 		return err
 	}
-	const sql = `
-INSERT INTO fs.inode VALUES ($1, $2);
-INSERT INTO fs.namespace VALUES ($3, $4, $1);
-`
-	if _, err := tx.Exec(sql, node.ID, inode, parentID, name); err != nil {
+	const insertNode = `INSERT INTO fs.inode VALUES ($1, $2)`
+	if _, err := tx.Exec(insertNode, node.ID, inode); err != nil {
+		_ = tx.Rollback()
+		return err
+	}
+	const insertNamespace = `INSERT INTO fs.namespace VALUES ($1, $2, $3)`
+	if _, err := tx.Exec(insertNamespace, parentID, name, node.ID); err != nil {
```

Comment:
I'm not sure this use of `RETURNING` is hyper-optimized. And I think we want to encourage these mechanisms so that they can see wider usage. Show of hands: who knew that `RETURNING` could be used in this way? I'm guessing the answer is low and I would attribute that in part to not seeing examples of it being used.

That said, I don't feel that strongly about using `RETURNING` here. It was just a suggestion.